### PR TITLE
Add Toolkit package

### PR DESCRIPTION
**Pull request by 0xGabi**
# 🦅 Pull Request

Remove `cli-utils` package and move it's logic inside `aragon-cli` and `create-aragon-app`.

Extract API from `aragon-cli` leaving all UI related functionality. Create a new package `toolkit` with the extracted API logic.

Move tests from `aragon-cli` to `toolkit`. 

I decided to wait before moving IPFS logic to the `toolkit` package. I was questioning if we want to have it there, or what logic on specific.

**TODO:**

- [x] Fix two tests on `extractContractInfoToFile` from `toolkit` package

### CODE REVIEW

File: packages/aragon-cli/src/lib/environment/configEnvironment.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 const Web3 = require('web3')
 const { merge } = require('lodash')
-
-const defaultEnvironments = require('../../config/environments.default')
-const defaultNetworks = require('../../config/truffle.default')
+//
```
Comment (ajsantander) [inline packages/aragon-cli/src/lib/environment/configEnvironment.js]: Typo?

File: packages/aragon-cli/src/lib/environment/loadConfigFiles.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
 const path = require('path')
 const fs = require('fs-extra')
-const { findProjectRoot } = require('../util')
 const Ajv = require('ajv')
-const arappSchema = require('../../schemas/arapp.schema')
+//
```
Comment (ajsantander) [inline packages/aragon-cli/src/lib/environment/loadConfigFiles.js]: Typo?

File: packages/aragon-cli/src/lib/node/process.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import execa from 'execa'
-import { withTimeout } from './misc'
 import killProcessOnPort from 'kill-port'
+//
```
Comment (ajsantander) [inline packages/aragon-cli/src/lib/node/process.js]: Typo?

File: packages/aragon-cli/test/acl/acl.test.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import test from 'ava'
-import { getDaoAddressPermissionsApps } from '../../src/lib/acl/view'
-import { getLocalWeb3, getNewDaoAddress } from './test-utils'
+import { getDaoAddressPermissionsApps } from '@aragon/toolkit/dist/acl/view'
+//
```
Comment (ajsantander) [inline packages/aragon-cli/test/acl/acl.test.js]: Typo?

File: packages/aragon-cli/test/apm/apmAclUtil.test.js [hunk 5]
```diff
@@ -1,7 +1,8 @@
 import test from 'ava'
-import ACL from '../../src/lib/apm/util/acl'
-import { getLocalWeb3 } from './test-utils'
-import { resolveEnsDomain } from '../../src/helpers/aragonjs-wrapper'
+import ACL from '@aragon/toolkit/dist/apm/util/acl'
+import { resolveEnsDomain } from '@aragon/toolkit/dist/helpers/aragonjs-wrapper'
+//
```
Comment (ajsantander) [inline packages/aragon-cli/test/apm/apmAclUtil.test.js]: Typo?

File: packages/aragon-cli/test/deploy/deploy.test.js [hunk 6]
```diff
@@ -1,7 +1,9 @@
 import test from 'ava'
 import web3EthAbi from 'web3-eth-abi'
-import { deployContract } from '../../src/lib/deploy'
-import { isAddress, isValidTxHash, getLocalWeb3 } from './test-utils'
+import { expandLink } from '@aragon/toolkit/dist/util'
+//
```
Comment (ajsantander) [inline packages/aragon-cli/test/deploy/deploy.test.js]: Ok, I'll stop reporting these :laughing: 
Maybe do a global search and replace?

File: packages/aragon-cli/test/ipfs/daemonAndConfig.test.js [hunk 7]
```diff
@@ -1,4 +1,6 @@
 import { serial as test } from 'ava'
+// import { remove, ensureDir } from 'fs-extra'
```
Comment (ajsantander) [inline packages/aragon-cli/test/ipfs/daemonAndConfig.test.js]: Remove commented out code?

**Comment by ajsantander** [comment]
@0xGabi outstanding work! This is a huge milestone :rocket: 

Regarding IPFS: I'd say that we can leave it in lib until we need the code elsewhere.

Regarding all your other items, I don't think that they are relevant for this particular task, which is to create the toolkit package and move things there. If we were to solve all those items before this is merged, I'd delay things significantly. That's why I always prefer to try to do a single, very specific, thing per PR. It makes the review process easier and faster. So, I'd say that you review all these points that you've made, and see if any of them need to be added to the v7 issue: https://github.com/aragon/aragon-cli/issues/981 - and we go ahead and merge this one :)

File: packages/aragon-cli/test/ipfs/daemonAndConfig.test.js (hunk 7 above)
Comment (0xGabi) [reply packages/aragon-cli/test/ipfs/daemonAndConfig.test.js]: I think @0x6431346e wanted to be sure to add those on a future PR. Is this still relevant?

File: packages/aragon-cli/test/deploy/deploy.test.js (hunk 6 above)
Comment (0xGabi) [reply packages/aragon-cli/test/deploy/deploy.test.js]: I saw Daniel did this kind of split between code imported from packages and code imported from internal files. What you think?

**Pull request merged**